Update caching.mdx (#83149)
vercel/next.js: docs/01-app/02-guides/caching.mdx

### `fetch`

Next.js extends the native `fetch` API to provide a powerful caching mechanism. The default caching behavior of `fetch` depends on how your route is rendered:

* **Static Rendering**: If a route is statically rendered (the default), `fetch` requests are cached in the **Data Cache**.

> * **Dynamic Rendering**: If a route is dynamically rendered, `fetch` requests are not cached. This behavior is equivalent to setting `{ cache: 'no-store' }`.

Reviewer: ```suggestion
Data returned from `fetch` is _not_ automatically cached in the Data Cache.

By default, when no `cache` or `next.revalidate` option is provided:

- With [static rendering](/docs/app/getting-started/partial-prerendering#static-rendering), the page is generated ahead of time (prerendered). The fetched data is stored in the [Data Cache](#data-cache) and the rendered output in the [Full Route Cache](#full-route-cache), and serves that result until the path is revalidated.
- With [dynamic rendering](/docs/app/getting-started/partial-prerendering#dynamic-rendering), the page is generated at request time. The fetch runs on every request and returns fresh data on each request.
```

I think, saying, this is equivalent to - in this case, but not the other, through zero changes to the fetch code as a dev, is often where the confusion begins, I'd say it is rather more "friendly" to say, there's two cases, and they behave like this...

Author: thanks

You can override this default behavior using the `cache` option:

> 

Reviewer: Maybe we can delete this and the snippet after wards?

Author: yep sure

Data returned from `fetch` is _not_ automatically cached in the Data Cache.

> By default, when no `cache` or `next.revalidate` option is provided:
> 
> - The fetched data is stored in the Data Cache and the rendered output is stored in the Full Route Cache. Next.js serves this cached result until the path is revalidated." or similar clearer phrasing.
> 
> - With [dynamic rendering](/docs/app/getting-started/partial-prerendering#dynamic-rendering), the page is generated at request time. The fetch runs on every request and returns fresh data on each request.

Reviewer: ```suggestion
By default, when no `cache` or `next.revalidate` option is provided:

- [Dynamic rendering](/docs/app/getting-started/partial-prerendering#dynamic-rendering): Fetch runs on every request and always returns fresh data.
- [Static rendering](/docs/app/getting-started/partial-prerendering#static-rendering): Fetched data is stored in the [Data Cache](#data-cache), and the rendered output in the [Full Route Cache](#full-route-cache). Next.js serves this cached result until the path is revalidated.
```
